com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/TimeProfiler.java

```diff
@@ -225,14 +197,25 @@ public static void createReport(Boolean fromEditor) {
                         System.out.println("Failed to create directories: " + parentDir);
                     }
                 }
-                if (extension.equals(".json")) {
-                    saveJSON(jsonReport, finalReportFile);
-                }
-                else if (extension.equals(".html")) {
-                    saveHTML(jsonReport, finalReportFile);
+                int bufferSize = 8 * 1024 * 1024; // 8 MB in bytes
+                try (BufferedWriter writer = new BufferedWriter(new FileWriter(finalReportFile), bufferSize)) {
```

Comment:
Create a bufferWriter that writes every 8 Mb on disk directly without creation of a huge String.